server/main.py

```diff
@@ -0,0 +1,131 @@
+import json
+import os
+import uuid
+from datetime import datetime
+from flask import Flask, request, jsonify
+
+from utilities import dir_utils
+
+DEFAULT_PORT = 8080
+
+# create an instance of the Flask class and assigns it to the variable app
+app = Flask(__name__)
+# set the configuration parameter 'UPLOAD_FOLDER' in app object to the value of UPLOAD_FOLDER
+app.config['UPLOAD_FOLDER'] = dir_utils.UPLOAD_FOLDER
+
+
+def generate_filename(uid, original_filename):
+    """
+        Generate a new filename by combining the original filename, timestamp, and UID.
+
+        Args:
+            uid (str): Unique identifier for the file.
+            original_filename (str): Original filename of the uploaded file.
+
+        Returns:
+            str: New filename with the format 'original_filename_timestamp_uid'.
+    """
+    timestamp = datetime.now().strftime('%Y%m%d%H%M%S')
+    return f"{''.join(original_filename.split('.')[:-1])}_{timestamp}_{uid}.{original_filename.split('.')[-1]}"
+
+
+def save_file(file, filename):
+    """
+       Save the uploaded file to the specified filename in the upload folder.
+
+       Args:
+           file (FileStorage): Uploaded file object.
+           filename (str): Filename to save the file as.
+    """
+    file.save(os.path.join(app.config['UPLOAD_FOLDER'], filename))
+
+
+def find_files(files, uid):
+    """
+        Find files that contain the given UID in the given list.
+
+        Args:
+            files (List[str]): List of filenames to search.
+            uid (str): Unique identifier to match.
+
+        Returns:
+            List[str]: List of filenames that contain the UID.
+    """
+    # the list comprehension iterates over files and adds a file if it contains the uid in its name
+    matching_files = [file for file in files if uid in file]
+    return matching_files
+
+
+@app.route('/upload', methods=['POST'])
+def upload():
+    """
+        Handle the upload endpoint.
+
+        This endpoint receives a POST request with an attached file, generates a UID for the uploaded file,
+        saves the file in the 'uploads' folder with a new filename containing the original filename, timestamp,
+        and UID, and returns a JSON object with the UID of the upload.
+
+        Returns:
+            Response: JSON response with the UID of the upload.
+    """
+    if 'file' not in request.files:
+        return jsonify({'error': 'No file attached'}), 400
```

Comment:
In this line you return a Tuple, and in your other return statement in the method you only return a jsonify object.